packages/calcite-components/src/components/tile/tile.tsx

```diff
@@ -214,6 +224,12 @@ export class Tile implements InteractiveComponent, SelectableComponent {
     this.calciteTileSelect.emit();
   };
 
+  private handleSlotChange = (event: Event): void => {
+    const slotNameCamelCased = camelCase((event.target as HTMLSlotElement).name);
```

Comment:
Could you use a `data-attribute` on `<slot>` with the corresponding state prop name instead of building it?